module/VuFind/src/VuFind/Db/Row/Ratings.php

```diff
@@ -49,4 +70,110 @@ public function __construct($adapter)
     {
         parent::__construct('id', 'ratings', $adapter);
     }
+
+    /**
+     * Get identifier (returns null for an uninitialized or non-persisted object).
+     *
+     * @return ?int
+     */
+    public function getId(): ?int
+    {
+        return $this->id ?? null;
+    }
+
+    /**
+     * Get user.
+     *
+     * @return ?UserEntityInterface
+     */
+    public function getUser(): ?UserEntityInterface
+    {
+        return $this->user_id
+            ? $this->getDbServiceManager()->get(UserServiceInterface::class)->getUserById($this->user_id)
+            : null;
+    }
+
+    /**
+     * Set user.
+     *
+     * @param ?UserEntityInterface $user User
+     *
+     * @return RatingsEntityInterface
+     */
+    public function setUser(?UserEntityInterface $user): RatingsEntityInterface
+    {
+        $this->user_id = $user?->getId();
+        return $this;
+    }
+
+    /**
+     * Get resource.
+     *
+     * @return ResourceEntityInterface
+     */
+    public function getResource(): ResourceEntityInterface
+    {
+        return $this->resource_id
+        ? $this->getDbServiceManager()->get(ResourceServiceInterface::class)->getResourceById($this->resource_id)
+        : null;
+    }
+
+    /**
+     * Set resource.
+     *
+     * @param ResourceEntityInterface $resource_id Resource
+     *
+     * @return RatingsEntityInterface
+     */
+    public function setResource(ResourceEntityInterface $resource_id): RatingsEntityInterface
+    {
+        $this->resource_id = $resource_id?->getId();
```

Comment:
You don't need to use the nullsafe operator (`?->`) here since the incoming value is not allowed to be null. I would also call the argument `$resource` instead of `$resource_id` since it is a full object, not just an ID.